main.go

```diff
@@ -213,6 +264,22 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Only start the controller when the leader election is won
+	mgr.Add(manager.RunnableFunc(func(stop <-chan struct{}) error {
+		setupLog.Info("starting custom controller")
+
+		stopChannel := make(chan struct{})
```

Comment:
i don't think we need to do an another stopChannel here. we should directly use the stop channel passed from controller. please sync with @abhipth on why it's needed.
also, we shouldn't close the stopChannel inside `StartController`, a component shouldn't close channel passed to it.